pkg/operator/controller/gateway-service-dns/controller.go

```diff
@@ -254,6 +275,18 @@ func (r *reconciler) ensureDNSRecordsForGateway(ctx context.Context, gateway *ga
 	return errs
 }
 
+// domainsEqual compares 2 DNS domains and returns if they are equivalent. Domains will be compared as if they were
+// fully qualified (ending with ".").
```

Comment:
The trailing dot should already be on the gateway hostname.  https://github.com/rfredette/cluster-ingress-operator/blob/master/pkg/operator/controller/gateway-service-dns/controller.go#L219-L221